dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageErrorMatcher.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+/** Matcher for linkage errors. A linkage error has a target symbol and a source class file. */
+class LinkageErrorMatcher implements SymbolProblemMatcher {
+
+  /** Matcher for the source class of the linkage error */
+  private SourceMatcher sourceMatcher;
+
+  /** Matcher for the target symbol of the linkage error */
+  private TargetMatcher targetMatcher;
+
+  void setSourceMatcher(SourceMatcher sourceMatcher) {
+    this.sourceMatcher = sourceMatcher;
+  }
+
+  void setTargetMatcher(TargetMatcher targetMatcher) {
+    this.targetMatcher = targetMatcher;
+  }
+
+  /**
+   * Returns true if all of non-null matchers of {@link #sourceMatcher} and {@link #targetMatcher}
```

Comment:
all of --> all
Can there be null matchers?

Reply:
No `NullMatcher` but `sourceMatcher` or `targetMatcher` field can be null.

Reply:
Removed "of" in "all of"